hummingbot/core/api_throttler/varied_rate_api_throttler.py

```diff
@@ -0,0 +1,54 @@
+import asyncio
+
+from typing import Deque, List
+
+from hummingbot.core.api_throttler.api_request_context_base import APIRequestContextBase
+from hummingbot.core.api_throttler.api_throttler_base import APIThrottlerBase
+from hummingbot.core.api_throttler.data_types import (
+    RateLimit,
+    Seconds,
+    TaskLog
+)
+
+
+class VariedRateThrottler(APIThrottlerBase):
+
+    def __init__(self,
+                 rate_limit_list: List[RateLimit],
+                 period_safety_margin: Seconds = 0.1,
+                 retry_interval: Seconds = 0.1):
+        super().__init__(rate_limit_list, period_safety_margin=period_safety_margin, retry_interval=retry_interval)
+
+    def execute_task(self, path_url: str):
+        rate_limit: RateLimit = self._path_rate_limit_map[path_url]
+
+        return VariedRateRequestContext(
+            task_logs=self._task_logs,
+            rate_limit=rate_limit,
+            stop_event=self._throttler_stop_event,
+            period_safety_margin=self._period_safety_margin,
+            retry_interval=self._retry_interval,
+        )
+
+
+class VariedRateRequestContext(APIRequestContextBase):
+
+    def __init__(self,
```

Comment:
`__init__` here just calls super(). It is not required.

Reply:
For `VariedRateRequestContext` I have decided that some additional variables be included.